### PR TITLE
ENH: add layout metadata compiler and examples

### PR DESCRIPTION
**Pull request by SantiagoTorres**
Hello, this commit adds the metadata compiler and a couple of metadata samples (debian grep and seattle).
Don't review the link metadata files, just the readme and the compiler please.

### CODE REVIEW

**Comment by lukpueh** [comment]
Btw. here's a list of required metadata schema changes:

- metablock
  - all top level fields are now under the `signed` field, and the `signatures` field is a sibling of the `signed` field
- layout
  - `expected_command` and `run` must be lists
  - `expires` must not have milliseconds
  - `material_matchrules` are now `expecte_materials`
  - `product_matchrules` are now `expecte_products`
  - `MATCH` rule syntax has change
- link
  - `_type` value is lowercase
  - `return_values` is now part of `byproducts`

**Comment by SantiagoTorres** [comment]
Hi!

> I wonder if it's worth building this into in-toto? AFAICS, the larger part of the script takes care of traversing the metadata, in order to sort fields and truncate long values, which largely aligns with the feature request in in-toto/in-toto#18. What do you think?

I'm not entirely sure if this is what's to be addressed on this side. I intentionally avoided using in-toto as a dependency (or any templating library for that matter) so as to keep this script-y. I believe that that issue has been floating around and having different meanings every time we revisit it, I'm afraid.

As for the point issues:
> The script is not compatible with the current metadata specification (nor are the ~30K added lines of metadata)
Yes, unfortunately time has gone by, we may want to update it to conform to the latest spec.

> It shouldn't be required that the layout is called root.layout

Probably not, but considering we're the ones that will be running this to update our examples in the docs I don't see why we should care much about user interface right away (this could be also ticketized)

> Links are loaded by globbing for *.link files and showed in the order glob returns them. Should we not rather load them as they are defined in layout, as e.g. in_toto.verifylib.load_links_for_layout does?

This requires a dependency on in-toto or smarter parsing of json objects, which I tried to avoid. Again, we could make the tool smarter if we'd like by adding more deps/code, but we may want to think about it for just a docs repo.
 
> Link file globbing did not work for me (I didn't troubleshoot though)

I suspect it's because the keyid prefix,

> Sublayout's are not handled

No, as we don't have any examples on the docs repo that use sublayouts. We can always add support for this as the need arises. 

> IMO it comes unexpected that the displayed materials and products are random samples if there are more than 9 of them

I can't personally think of any other way to keep things succint, but I'm open to suggestions. I do agree this is *not* a perfect solution to overly verbose metadata.

> Adding ellipses as last item (to show that a dict is truncated), does not guarantee that it is indeed printed as last item. Python does not guarantee to keep the order of dict items as they were inserted.

This is true, and it's something I bailed on working on back then. We could use a frozendict or serialize and then append on the printout after-the-fact (which would be messy).

> A bare minimum of documentation would be nice

Agreed. Let's decide on whether this goes here and work accordingly.

**Comment by lukpueh** [comment]
@SantiagoTorres, I updated the sample layout and link metadata to meet the latest version of the spec, cleaned-up the compiler script, and used it to create new versions of the markdown-formatted summaries, which I have already published on our website (see [pph](https://in-toto.github.io/examples/polypasswordhasher.html), [seattle](https://in-toto.github.io/examples/seattle.html) and [debian](https://in-toto.github.io/examples/debian.html)).

Let me know what you think.

**Comment by lukpueh** [comment]
Btw. 088277b is my stab at in-toto/in-toto#18. IMO especially the dict field ordering for pretty printing is very useful.

File: examples/compile-examples.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+def _beautify(obj, **kw):
+  """Recurisve helper for beautify. See beautify for more info. """
+
+  # Truncate strings
+  if isinstance(obj, six.string_types):
+    if len(obj) > kw["max_str_len"]:
+      obj = obj[:kw["max_str_len"] - 3] + "..."
+
+  # Truncate list and recurse into _beautify for each  item
```
Comment (adityasaky) [inline examples/compile-examples.py]: Extra space after each

Comment (lukpueh) [reply examples/compile-examples.py]: Good eyes! Thanks for the review. :)

**Pull request merged**